connectors/jdbc/translator-jdbc/src/main/java/org/teiid/translator/jdbc/exasol/ExasolExecutionFactory.java

```diff
@@ -0,0 +1,405 @@
+        addPushDownFunction(EXASOL, ADD_WEEKS, DATE, DATE, INTEGER);
+        addPushDownFunction(EXASOL, ADD_WEEKS, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, ADD_YEARS, DATE, DATE, INTEGER);
+        addPushDownFunction(EXASOL, ADD_YEARS, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, DATE_TRUNC, DATE, STRING, DATE);
+        addPushDownFunction(EXASOL, DATE_TRUNC, TIMESTAMP, STRING, TIMESTAMP);
+        addPushDownFunction(EXASOL, DAYS_BETWEEN, INTEGER, DATE, DATE);
+        addPushDownFunction(EXASOL, DAYS_BETWEEN, INTEGER, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, HOURS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, MINUTES_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, MONTHS_BETWEEN, DOUBLE, DATE, DATE);
+        addPushDownFunction(EXASOL, MONTHS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, POSIX_TIME, DOUBLE, TIMESTAMP);
+        addPushDownFunction(EXASOL, ROUND, DATE, DATE);
+        addPushDownFunction(EXASOL, ROUND, DATE, DATE, STRING);
+        addPushDownFunction(EXASOL, ROUND, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, ROUND, TIMESTAMP, TIMESTAMP, STRING);
+        addPushDownFunction(EXASOL, SECONDS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, TRUNC, DATE, DATE);
+        addPushDownFunction(EXASOL, TRUNC, DATE, DATE, STRING);
+        addPushDownFunction(EXASOL, TRUNC, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, TRUNC, TIMESTAMP, TIMESTAMP, STRING);
+        addPushDownFunction(EXASOL, TRUNCATE, DATE, DATE);
+        addPushDownFunction(EXASOL, TRUNCATE, DATE, DATE, STRING);
+        addPushDownFunction(EXASOL, TRUNCATE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, TRUNCATE, TIMESTAMP, TIMESTAMP, STRING);
+        addPushDownFunction(EXASOL, YEARS_BETWEEN, DOUBLE, DATE, DATE);
+        addPushDownFunction(EXASOL, YEARS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+
+        registerFunctionModifier(SourceSystemFunctions.FROM_UNIXTIME, new AliasModifier(FROM_POSIX_TIME));
+        registerFunctionModifier(SourceSystemFunctions.DAYOFMONTH, new AliasModifier(DAY));
+        registerFunctionModifier(SourceSystemFunctions.CURDATE, new AliasModifier(UPPER));
+        registerFunctionModifier(SourceSystemFunctions.FORMATTIMESTAMP, new AliasModifier(TO_CHAR));
+        registerFunctionModifier(SourceSystemFunctions.PARSETIMESTAMP, new AliasModifier(TO_TIMESTAMP));
+
+        //Convert
+        ConvertModifier convertModifier = new ConvertModifier();
+
+        convertModifier.addTypeMapping("VARCHAR(4000)", FunctionModifier.STRING); 
+        convertModifier.addTypeMapping("DECIMAL(3)", FunctionModifier.BYTE);
+        convertModifier.addTypeMapping("DECIMAL(5)", FunctionModifier.SHORT);
+        convertModifier.addTypeMapping("DECIMAL(18)", FunctionModifier.INTEGER);
```

Comment:
Should this be DECIMAL(10) for INTEGER?

If needed we can also have the useIntegralTypes property drive the supportsConvert method to prevent integral conversions from being pushed down.

Reply:
So they actually have an alias for Integer and mapped it to DECIMAL(18)... It's page 108 in their manual at https://www.exasol.com/portal/display/DOC/User+Manual+6.0?preview=/20056178/33096816/EXASOL_User_Manual-6.0.8-en.pdf